File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int l=0, res= nums[0], r= nums.size()-1;
        while(l<=r){
            int mid= (r-l)/2 + l;
            res = min(res, nums[mid]);
            if(nums[mid] >= nums[l]){
                res = min(res, nums[l]);
                l = mid+1;
            }
            else{
                r = mid-1;
            }
        }
        return res;
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l , r = 0, len(nums)-1
        res = nums[0]
        
        while l <= r :
            mid = (l+r)//2
            res = min(nums[mid], res)
            
            if nums[l] <= nums[mid]:
                res = min(nums[l], res)
                l = mid + 1
            else:
                r = mid - 1
                
        return res
        
        